packages/protocol/contracts/governance/Governance.sol

```diff
@@ -672,22 +773,50 @@ contract Governance is
       // Skip proposals where there was no vote cast by the user AND
       // ensure vote record proposal matches identifier of dequeued index proposal.
       if (
-        voteRecord.value != Proposals.VoteValue.None &&
+        (voteRecord.weights.length > 0 || voteRecord.deprecated_weight > 0) &&
```

Comment:
I think the natspec of this function should reflect all votes are revoked. We could also create a method to allow user to revoke votes partially, may make stCelo implementation simpler in the future.

Reply:
I'm not sure what exactly do you mean :-) Function natspec is stating "Revoke votes on all proposals of sender in the referendum stage." - would you like to reword it somehow ?

Regarding new method to revoke votes - currently it is possible to to call votePartially with empty voteValues (and weights) which will effectively revoke votes for particular proposal.